numpy/core/src/common/ufunc_override.c

```diff
@@ -30,6 +33,12 @@ PyUFuncOverride_GetNonDefaultArrayUfunc(PyObject *obj)
     if (PyArray_CheckExact(obj)) {
         return NULL;
     }
+   /* Fast return for numpy scalar types */
+    if (is_anyscalar_exact(obj) )
+    {
```

Comment:
```suggestion
    if (is_anyscalar_exact(obj)) {
```
Style nit, but mabe might as well merge it with the array check above?  If you use this helper, I am tempted to rename it, but it doesn't matter.